apps/ejabberd/src/ejabberd_local.erl

```diff
@@ -88,15 +86,15 @@
 start_link() ->
     gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
 
--spec process_iq(From :: ejabberd:jid(),
+-spec process_iq(Acc :: mongoose_acc:t(),
+                 From :: ejabberd:jid(),
                  To :: ejabberd:jid(),
-                 Acc :: mongoose_acc:t()
+                 El :: xmlel()
                  ) -> 'nothing' | 'ok' | 'todo' | pid()
                     | {'error', 'lager_not_running'} | {'process_iq', _, _, _}.
-process_iq(From, To, Acc0) ->
+process_iq(Acc0, From, To, El) ->
     Acc = mongoose_acc:require(iq_query_info, Acc0),
     IQ = mongoose_acc:get(iq_query_info, Acc),
-    El = mongoose_acc:get(element, Acc),
```

Comment:
I don't understand the question. In Acc we have what came in, and we don't want to change it. When we want to send something out - which may be entirely different thing - we pass it as arg. Hence this change.